LineApp/src/main/java/org/example/Main.java

```diff
@@ -1,30 +1,93 @@
+            "t",
+            "type",
+            true,
+            "Set the type of the request"
+    );
+
+    private static void help(Options options) {
+        HelpFormatter hf = new HelpFormatter();
+        PrintWriter pw = new PrintWriter(System.out);
+        pw.println("LineApp");
+        pw.println();
+        hf.printUsage(pw, 100, "[options] owner repo key type");
+        hf.printOptions(pw, 100, options, 2, 5);
+        pw.close();
+    }
+
     public static void main(String[] args) {
-        //for users (line interface)
-        Scanner sc = new Scanner(System.in);
-        System.out.println("repo: 1,  issues: 2, pulls: 3");
-        String choice = sc.next();
-        Scanner sc2 = new Scanner(System.in);
-        System.out.println("owner, repo");
-        String ownerName = sc2.next();
-        String ownerRepo = sc2.next();
-        //gh api client obj
-        GitHubApiClient ghac = new GitHubApiClient(ownerName, ownerRepo);
-        //your choice what we get
-        if (choice.equals("1")) {
-            System.out.println(ghac.getRepo());
-        } else if (choice.equals("2")) {
-            System.out.println(ghac.getRepoIssue());
-        } else {
-            System.out.println(ghac.getRepoPulls());
+        CommandLineParser clp = new DefaultParser();
+        Options options = new Options();
+        options.addOption(ARG_OWNER);
+        options.addOption(ARG_REPO);
+        options.addOption(ARG_KEY);
+        options.addOption(ARG_TYPE);
+        try {
+            CommandLine cl = clp.parse(options, args);
+            String owner = null;
+            String repo = null;
+            String type = null;
+            String key = "";
+            try {
+                owner = cl.getOptionValue("owner");
+                repo = cl.getOptionValue("repo");
+                type = cl.getOptionValue("type");
+                key = cl.getOptionValue("key");
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+            //checks null references
+            if (owner == null || repo == null) {
+                help(options);
+                System.exit(-1);
+            }
+            //type: repo request by default
+            if(type == null){
+                type = "repo";
+            }
+            //call apiclient methods
+            GitHubApiClient gitHubApiClient = new GitHubApiClient(owner, repo);
+            if (type.equals("repo")) {
+                System.out.println(gitHubApiClient.getRepo());
+            } else if (type.equals("issues")) {
+                System.out.println(gitHubApiClient.getRepoIssue());
+            } else if (type.equals("pulls")) {
+                System.out.println(gitHubApiClient.getRepoPulls());
+            } else {
+                System.out.println("choose type repo, issues, pulls");
+                help(options);
+                System.exit(-1);
+            }
+        } catch (ParseException e) {
```

Comment:
looks like you meants to catch RuntimeException here and ParseException on line 61?